### PR TITLE
fix: pile drip

### PR DESCRIPTION
**Pull request by manuelpolzhofer**
implements #14

### CODE REVIEW

File: src/pile.sol [hunk 1]
```diff
@@ -129,15 +129,21 @@ contract Pile is DSNote {
         // --- Fee Accumulation ---
     function drip(uint fee) public {
         uint48 rho = fees[fee].rho;
+        require(now >= rho);
         uint speed = fees[fee].speed;
+        if (speed == 0) {
```
Comment (lucasvo) [inline src/pile.sol]: Does it breafk if we not have this check? It will be more easily verifiable this way.

File: src/pile.sol [hunk 2]
```diff
@@ -129,15 +129,21 @@ contract Pile is DSNote {
...
         uint chi = fees[fee].chi;
         uint debt = fees[fee].debt;
-        require(now >= rho);
-        uint chi_ = sub(rmul(rpow(speed, now - rho, ONE), chi), chi);
-        uint wad = mul(debt, chi_);
-        add(Debt, wad);
+
+        uint latest = rmul(rpow(speed, now - rho, ONE), chi);
+        uint chi_ = rdiv(latest, chi);
+        uint wad = rmul(debt, chi_)-debt;
```
Comment (lucasvo) [inline src/pile.sol]: I don't think this should be rmul but regular mul.

Comment (manuelpolzhofer) [reply src/pile.sol]: It needs to be `rmul`. I just tested it.

File: src/pile.sol (hunk 1 above)
Comment (manuelpolzhofer) [reply src/pile.sol]: Currently, it would break. If the `fee` is not set for a loan. but I think I can avoid it.

Comment (lucasvo) [reply src/pile.sol]: Does it produce bad values or does it divide by zero? I think a revert here is fine if it's not set.

Comment (manuelpolzhofer) [reply src/pile.sol]: The admit method will get a new parameter `fee` afterwards `speed==0` won't happen anymore.

File: src/pile.sol [hunk 3]
```diff
@@ -61,6 +61,8 @@ contract Pile is DSNote {
     constructor(address tkn_) public {
         wards[msg.sender] = 1;
         tkn = TokenLike(tkn_);
+        fees[0].chi = ONE;
+        fees[0].speed = ONE;
```
Comment (manuelpolzhofer) [inline src/pile.sol]: @lucasvo solves the problem

File: src/pile.sol (hunk 1 above)
Comment (lucasvo) [reply src/pile.sol]: I'd prefer to leave it as a separate method and add a call to the reception that does both at the same time.

Comment (manuelpolzhofer) [reply src/pile.sol]: The reception is only called by the borrower. I think it makes sense to pass it in the admit method and the admit contract calls `pile.file` afterwards.

**Pull request merged**